File: 2022/Day1.py
#! /usr/bin/env python
# pylint: disable=unused-wildcard-import
from util import *

def parse(inp):
    return sorted([sum(nums(x)) for x in superlines(inp)])

def p1(inp):
    return inp[-1]

def p2(inp):
    return sum(inp[-3:])

def main():
    inp = parse(data())
    print(f"Solution for part 1:\n{p1(inp)}")
    print(f"Solution for part 2:\n{p2(inp)}")

if __name__ == "__main__":
    main()

# Solution part 1: 67450
# Solution part 2: 199357
# Leaderboard: 882 / 605

File: 2022/Day1.hs
#! /usr/bin/env -S stack runhaskell
import Data.List
import Data.List.Split

parse :: String -> [Int]
parse = reverse . sort . map (sum . map read . lines) . splitOn "\n\n"

part1 :: [Int] -> Int
part1 = head

part2 :: [Int] -> Int
part2 = sum . take 3

main :: IO()
main = do
    f <- readFile("1.in")
    let input = parse f
    putStr "Solution for part 1: "
    putStrLn . show $ part1 input
    putStr "Solution for part 2: "
    putStrLn . show $ part2 input

-- Solution part 1: 67450
-- Solution part 2: 199357
